src/app/clusters/energy-evse-server/EnergyEvseTestEventTriggerDelegate.h

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/TestEventTriggerDelegate.h>
+
+namespace chip {
+
+enum class EnergyEvseTrigger : uint64_t
+{
+    // Scenarios
+    kBasicFunctionality = 0x0099000000000000,      // Basic Functionality Test Event       | Simulate installation with
+                                                   // _{A_CIRCUIT_CAPACITY}_=32A and _{A_USER_MAXIMUM_CHARGE_CURRENT}_=32A
+    kBasicFunctionalityClear = 0x0099000000000001, // Basic Functionality Test Event Clear | End simulation of installation
+    kEVPluggedIn             = 0x0099000000000002, // EV Plugged-in Test Event | Simulate plugging
+                                                   // the EV into the EVSE using a cable of 63A capacity
+    kEVPluggedInClear    = 0x0099000000000003,     // EV Plugged-in Test Event Clear | Simulate unplugging the EV
+    kEVChargeDemand      = 0x0099000000000004, // EV Charge Demand Test Event | Simulate the EV presenting charge demand to the EVSE
+    kEVChargeDemandClear = 0x0099000000000005, // EV Charge Demand Test Event Clear | Simulate the EV becoming fully charged
+
+};
+
+class EnergyEvseTestEventTriggerDelegate : public TestEventTriggerDelegate
+{
+public:
+    explicit EnergyEvseTestEventTriggerDelegate(const ByteSpan & enableKey, TestEventTriggerDelegate * otherDelegate) :
```

Comment:
So this relies on the caller keeping the buffer `enableKey` points to alive as long as this object is alive, right?  At minimum, this needs to be documented.  But could also just make a local copy into a member instead...

Reply:
Again just copied the pattern from SmokeCOTest and OTATestEvent Trigger models.
So the same comment here would apply to them as well.

Reply:
The other code may well be broken too....

Reply:
In terms of implementation not sure if you've found a security hole here?
When you run the application like example/energy-management app on linux, you need to add --enable-key arg with the magic secret key.

In https://github.com/project-chip/connectedhomeip/blob/a50f95964f2162f969a5e4e4f0b49a5ea1f0d998/examples/platform/linux/AppMain.cpp#L324-L369 there is a SampleTestEventTriggerDelegate class.

A bit later these other test event handlers are compiled into the application and each handler (one per cluster) gets added to a chain to be checked to see if it handles the Trigger, but in Linux at least I think it is handled by the SampleTestEventTrigger doing the checking of the enable key. However if that sample wasn't there then this implementation (like src/app/clusters/smoke-co-alarm-server/SmokeCOTestEventTriggerDelegate.cpp and src/app/clusters/ota-requestor/OTATestEventTriggerDelegate.cpp) doesn't have an Init().

See later in the same file:
https://github.com/project-chip/connectedhomeip/blob/a50f95964f2162f969a5e4e4f0b49a5ea1f0d998/examples/platform/linux/AppMain.cpp#L542-L560

Other platforms seem to have a static enableKey which may come from the factory data provider:
e.g. 
https://github.com/project-chip/connectedhomeip/blob/a50f95964f2162f969a5e4e4f0b49a5ea1f0d998/examples/light-switch-app/infineon/cyw30739/src/main.cpp#L64-L66

Reply:
My point is that nothing in the API makes it clear that the thing passed in needs to have "lives forever" lifetime.  Yes, current simple things we have use "forever" lifetimes because it's convenient, but nothing prevents someone writing an app from loading this thing from flash and passing it in... and then dropping the buffer.  And nothing documents they shouldn't do that.

Reply:
I've added a comment to the header to help document this point.